napari/utils/migrations.py

```diff
@@ -211,3 +211,43 @@ def __init_subclass__(cls, **kwargs):
     _OldClass.__new__.__signature__ = prealloc_signature  # type: ignore [attr-defined]
 
     return _OldClass
+
+
+class _DeprecatingDict(dict[str, Any]):
```

Comment:
Unsure if this is the best way to sub-class a `dict` (e.g. should I extend a `MutableMapping` instead and implement that?). Open to other ideas.